lib/ts/styles/styleContext.tsx

```diff
@@ -74,7 +74,8 @@ export function StyleProvider({
  * Helpers
  */
 function getMergedPalette(defaultPalette: NormalisedPalette, rawPalette: Record<string, string>): NormalisedPalette {
-    const palette = defaultPalette;
+    // We copy the defaultPalette into in order to not update the original
+    const palette = { colors: { ...defaultPalette.colors }, fonts: { ...defaultPalette.fonts } };
```

Comment:
What if the `defaultPalette` has something new later? Why not do a generic deep clone?

Reply:
We'd have to add it here, but (and update the same kind of thing in a bunch of other places). We didn't have a generic deep copy util, so I didn't want to add one just for this case, because it changes very rarely.

Reply:
Okay